src/common/feedGenerator.ts

```diff
@@ -56,31 +57,52 @@ export const whereKeyword = (
   return `EXISTS${query}`;
 };
 
+export const getExcludedAdvancedSettings = async (
+  con: ORMConnection,
+  feedId: string,
+  userId: string,
+): Promise<number[]> => {
+  const [features, advancedSettings, feedAdvancedSettings] = await Promise.all([
+    fetchUserFeatures(userId),
+    con.getRepository(AdvancedSettings).find(),
+    con.getRepository(FeedAdvancedSettings).find({ feedId, enabled: false }),
+  ]);
+  const userSettings = feedAdvancedSettings.reduce(
+    (obj, settings) => ({
+      ...obj,
+      [settings.advancedSettingsId]: settings.enabled,
+    }),
+    {},
+  );
+  const feature = features?.advanced_settings_default_values;
+  const excludedSettings = advancedSettings.filter((adv) => {
```

Comment:
Just a global note:
Finding this whole function super hard to read, is there any way we can abstract it a bit? 

It's super hard to see what the 3 values are, and what we actually return from here.

Reply:
If it was a bit hard to read for someone then it can be for the others as well. Let me find a way on how we can make this much simpler to read. Thanks for the note!